### PR TITLE
Updated Kafka security configuration

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Updated Kafka security configuration.
1. Modified to allow authentication and encryption configuration to be separate so that it can extended easily later
2. Added region and sts role arn to aws config.
3. Modified to set MSK IAM properties in the kafka configuration when that auth mechanism is used.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaSourceConfig.java [hunk 1]
```diff
@@ -19,6 +19,21 @@
  */
 
 public class KafkaSourceConfig {
+    public class SslConfig {
+        // TODO Add support for SSL Encryption by having
+        // path to certificates, etc
+    }
+
+    public class EncryptionConfig {
+        // Absence of ssl means no-encryption
```
Comment (hshardeesi) [inline data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaSourceConfig.java:28]: So this means PLAINTEXT? should we make it explicitly configurable. Customers may assume SSL by default if not specified.

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/AuthConfig.java [hunk 2]
```diff
@@ -5,33 +5,65 @@
...
-        this.plainTextAuthConfig = plainTextAuthConfig;
+    public SaslAuthConfig getSaslAuthConfig() {
+        return saslAuthConfig;
     }
 
-    public PlainTextAuthConfig getPlainTextAuthConfig() {
-        return plainTextAuthConfig;
+    @AssertTrue(message = "Only one of SSL or SASL auth config must be specified")
+    public boolean hasSaslOrSslConfig() {
+        return Stream.of(sslAuthConfig, saslAuthConfig).filter(n -> n!=null).count() == 1;
```
Comment (hshardeesi) [inline data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/AuthConfig.java:66]: should we have a similar check on sasl_mechanism as well? only one of the mechanisms should be specified.

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/AwsConfig.java [hunk 3]
```diff
@@ -10,10 +10,26 @@
...
-        return awsMskArn;
+    @Size(min = 1, message = "Region cannot be empty string")
+    @JsonProperty("region")
+    private String region;
+
+    @Size(min = 20, max = 2048, message = "sts_role_arn length should be between 20 and 2048 characters")
+    @JsonProperty("sts_role_arn")
+    private String stsRoleArn;
+
+    public String getMskArn() {
```
Comment (hshardeesi) [inline data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/AwsConfig.java:24]: We will most likely need to add an option to specify what type of bootstrap servers we want to read from MSK (public, single-vpc or multi-VPC) we can probably add that later also.

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaSourceConfig.java (hunk 1 above)
Comment (kkondaka) [reply data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaSourceConfig.java:28]: But SSL requires additional config like pointers to certificates, etc. Where as no-encryption does not require any config, right?

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/AwsConfig.java (hunk 3 above)
Comment (kkondaka) [reply data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/AwsConfig.java:24]: Isn't that decided on the MSK server side? Does client have option to read public if the server is not public?

Comment (kkondaka) [reply data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/AwsConfig.java:24]: Will add this in the next PR.

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaSourceConfig.java (hunk 1 above)
Comment (kkondaka) [reply data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaSourceConfig.java:28]: Defaulted to SSL.

**Review by hshardeesi — APPROVED** [review]

**Pull request merged**